standardtracer/tracer.go

```diff
@@ -2,16 +2,67 @@ package standardtracer
 
 import (
 	"sync"
+	"sync/atomic"
 	"time"
 
 	opentracing "github.com/opentracing/opentracing-go"
 )
 
-// New creates and returns a standard Tracer which defers to `recorder` after
-// RawSpans have been assembled.
-func New(recorder SpanRecorder) opentracing.Tracer {
+type sampleFunc func(int64) bool
+
+// Options allows creating a customized Tracer via NewWithOptions. The object
+// is thread safe and can be updated freely to influence the behavior of the
+// Tracer.
+type Options struct {
+	shouldSample       atomic.Value // contains a sampleFunc
+	trimUnsampledSpans int32        // updated atomically
+	recorder           atomic.Value // contains a SpanRecorder
```

Comment:
@bensigelman @tschottdorf why don't we fork it now into its own repo, like `reference-tracer-go`?

It will also help with vendoring, I am working on adding tracing instrumentation to our internal libs, which requires me to vendor opentracing, and I don't really need a standardtracer in there.